diesel/src/r2d2.rs

```diff
@@ -75,9 +75,31 @@ impl ::std::error::Error for Error {
     }
 }
 
+/// A trait indicating a connection could be used inside a r2d2 pool
+pub trait R2D2Connection: Connection {
+    /// Query used to check if a connection is still alive
+    ///
+    /// Should result in a database error if that's not the case
+    const CHECK_QUERY_STRING: &'static str;
```

Comment:
If `SELECT 1` isn't broad enough, should we just drop all assumptions and have this be `fn ping(&self) -> QueryResult<()>`?

Reply:
That would probably be the even better solution because this would theoretically allow to use sql independent connection checks provided be the libraries below.